web/src/app/components/tasks-editor/tasks-editor.component.ts

```diff
@@ -133,13 +139,13 @@ export class TasksEditorComponent {
     return this.formGroup.get('tasks') as FormArray;
   }
 
-  onTaskAdd(group: TaskGroup) {
+  onTaskAdd(group: TaskGroup, required = false) {
```

Comment:
Nit: This method was originally only meant to handle user "add task" events. Perhaps it would be clearer if you refactored the logic into an `addTask` method which is called both by `onTaskAdd` and `ngOnInit`.

Reply:
This comment no longer applies due to the new changes if approved.